cmd/object-api-utils.go

```diff
@@ -800,6 +800,11 @@ func NewGetObjectReader(rs *HTTPRangeSpec, oi ObjectInfo, opts ObjectOptions, cl
 			return r, nil
 		}
 	}
+	
+	if oi.Size == 0 {
```

Comment:
This seems like the wrong place to fix this. The length *is* 0, so it should be returned as so.

It seems more like it should be fixed where the header is calculated/added and maybe the call should be dropped completely if we know the object exists at this point.